auction-server/src/auction/service/auction_manager.rs

```diff
@@ -393,67 +387,103 @@ impl AuctionManager<Svm> for Service<Svm> {
+                    .transaction
+                    .signatures
+                    .first()
+                    .expect("Signature array is empty on svm bid tx")
+            })
+            .collect();
+        let statuses: Vec<_> = self
             .config
             .chain_config
             .client
-            .get_signature_status_with_commitment(
-                &bid_status_auction.tx_hash,
-                CommitmentConfig::confirmed(),
-            )
-            .await?;
+            .get_signature_statuses(&signatures)
+            .await?
+            .value
+            .into_iter()
+            .map(|status| {
+                status
+                    .filter(|status| status.satisfies_commitment(CommitmentConfig::confirmed()))
+                    .map(|status_meta| status_meta.status)
+            })
+            .collect();
 
-        let status = match status {
-            Some(res) => match res {
-                Ok(()) => entities::BidStatusSvm::Won {
-                    auction: bid_status_auction,
-                },
-                Err(_) => entities::BidStatusSvm::Failed {
-                    auction: bid_status_auction,
-                },
-            },
-            None => {
-                // not yet confirmed
-                // TODO Use the correct version of the expiration algorithm, which is:
-                // the tx is not expired as long as the block hash is still recent.
-                // Assuming a certain block time, the two minute threshold is good enough but in some cases, it's not correct.
-                if bids[0].initiation_time + BID_MAXIMUM_LIFE_TIME_SVM < OffsetDateTime::now_utc() {
-                    // If the bid is older than the maximum lifetime, it means that the block hash is now too old and the transaction is expired.
-                    entities::BidStatusSvm::Expired {
-                        auction: bid_status_auction,
+        let res = statuses
+            .iter()
+            .map(|status| {
+                let bid_status_auction = bid_status_auction.clone();
+
+                match status {
+                    Some(res) => Some(match res {
+                        Ok(()) => entities::BidStatusSvm::Won {
+                            auction: bid_status_auction,
+                        },
+                        Err(_) => entities::BidStatusSvm::Failed {
+                            auction: bid_status_auction,
+                        },
+                    }),
+                    None => {
+                        // not yet confirmed
+                        // TODO Use the correct version of the expiration algorithm, which is:
+                        // the tx is not expired as long as the block hash is still recent.
+                        // Assuming a certain block time, the two minute threshold is good enough but in some cases, it's not correct.
+                        if bids[0].initiation_time + BID_MAXIMUM_LIFE_TIME_SVM
```

Comment:
I think we should check the bids[i] for the expiration of that specific bid.

Reply:
good catch!